### PR TITLE
Pod logs refactoring

### PR DESCRIPTION
**Pull request by aleksfront**
Refactoring log components by spreading `PodLogs` monolith to logical and presentational ones.

Also fixing founded bugs including:
* Unclear appearing of jump to bottom button
* Loading more and more logs when switching tabs
* Log controls blinking

funbox logs

### CODE REVIEW

File: src/renderer/components/dock/pod-log-list.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+      return;
+    }
+    if (scrolledToBeginning && newLogsLoaded) {
+      this.virtualListDiv.current.scrollTop = (logs.length - prevProps.logs.length) * this.lineHeight;
+    }
+    if (fewLogsLoaded) {
+      this.isJumpButtonVisible = false;
+    }
+    if (!logs.length) {
+      this.isLastLineVisible = false;
```
Comment (Nokel81) [inline src/renderer/components/dock/pod-log-list.tsx:63]: This field really sounds like it is computed. Shouldn't it be marked as such?

File: src/renderer/components/dock/pod-log-list.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      pieces.forEach((piece, index) => {
+        const active = isActiveOverlay(rowIndex, index);
+        const lastItem = index === pieces.length - 1;
+        const overlayValue = matches.next().value;
+        const overlay = !lastItem ?
+          <span
+            className={cssNames("overlay", { active })}
+            dangerouslySetInnerHTML={{ __html: ansiToHtml(overlayValue) }}
+          /> :
+          null;
```
Comment (Nokel81) [inline src/renderer/components/dock/pod-log-list.tsx]: This format is rather confusing to read.
```suggestion
        const overlay = !lastItem
          ? <span
            className={cssNames("overlay", { active })}
            dangerouslySetInnerHTML={{ __html: ansiToHtml(overlayValue) }}
          />
          : null;
```

File: src/renderer/components/dock/pod-logs.tsx [hunk 3]
```diff
@@ -1,67 +1,34 @@
...
-  private colorConverter = new AnsiUp();
+  private logListElement = React.createRef<PodLogList>(); // A reference for VirtualList component
 
   componentDidMount() {
     disposeOnUnmount(this, [
       reaction(() => this.props.tab.id, async () => {
-        await this.load();
-        this.scrollToBottom();
+        await this.reload();
       }, { fireImmediately: true }),
```
Comment (Nokel81) [inline src/renderer/components/dock/pod-logs.tsx]: Couldn't we just pass `this.reload` to the reaction directly?

**Review by Nokel81 — APPROVED** [review]

File: src/renderer/components/dock/pod-logs.tsx (hunk 3 above)
Comment (aleksfront) [reply src/renderer/components/dock/pod-logs.tsx]: Can you elaborate this a bit please?

Comment (Nokel81) [reply src/renderer/components/dock/pod-logs.tsx]: Sure. It seems like we have the following snippet. `async () => { await this.reload(); }`. I wonder if we could instead just have, `this.reload` if it has been marked with `@autobind()`.

File: src/renderer/components/dock/pod-log-list.tsx (hunk 1 above)
Comment (aleksfront) [reply src/renderer/components/dock/pod-log-list.tsx:63]: I don't think so. This is an observable flag not really different from other ones like `isJumpButtonVisible`. A `computed` is used to calculate some value from other observables. `isLastLineVisible` just holds bool value and doesn't need to be "computed".

File: src/renderer/components/dock/pod-logs.tsx (hunk 3 above)
Comment (aleksfront) [reply src/renderer/components/dock/pod-logs.tsx]: Yes, it works as expected. Thanks for review!

**Pull request merged**